File: LeetCode75_LevelTwo/Insert_Interval/Insert_Interval.cc
// 57. Insert Interval C++
#include <vector>
using namespace std;

class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        vector<vector<int>> output;
        int start = newInterval[0];
        int end = newInterval[1];
        bool found = false;
        for (auto x: intervals) {
            if (found) {
                output.emplace_back(x);
                continue;
            }
            if (start > x[1]) {
                output.emplace_back(x);
            } else if (end < x[0]) {
                output.emplace_back( vector<int>{start, end});
                output.emplace_back(x);
                found = true;
            } else {
                start = min(start, x[0]);
                end = max(end, x[1]);
            }
        }
        if (!found) {
            output.emplace_back(vector<int>{start, end});
        }
        return output;
    }
};

File: LeetCode75_LevelTwo/Insert_Interval/Insert_Interval.py
# 57. Insert Interval Python
class Solution(object):
    def insert(self, intervals, newInterval):
        """
        :type intervals: List[List[int]]
        :type newInterval: List[int]
        :rtype: List[List[int]]
        """
        start = newInterval[0]
        end = newInterval[1]
        output = []
        found = False
        for l in intervals:
            if found:
                output.append(l)
                continue
            if l[1] < start:
                output.append(l)
            elif end < l[0]:
                output.append([start, end])
                output.append(l)
                found = True
            else:
                start = min(start, l[0])
                end = max(end, l[1])
        if not found:
            output.append([start, end])
        return output
